### ISSUE TITLE
phase-2 reco CPU hotspot: egammaTools::getConversionInfos

### ISSUE THREAD

**Issue by slava77** [cms-sw/cmssw#30954]
(pointed out by @dpiparo )

egammaTools::getConversionInfos is found to consume 1.1% of CPU time in workflow 23434.21 (step3), or about 1.6% of all reco module times (ignoring the I/O cost).
https://slava77sk.web.cern.ch/slava77sk/reco/cgi-bin/igprof-navigator/CMSSW_11_2_X_2020-07-21-2300-sign1106.376a0992c5d.23434.21.step3.50.pp/186

Things to improve: tighten code, delay in the if conditions deltaR which invokes math functions, no object instantiated in the loop, minimise function calls

 Some of the largest offending code in (repeated) computations of Track::eta and Track::phi.
For the eta/phi case, see recently implemented changes in RecoEgamma/EgammaElectronProducers/plugins/GEDGsfElectronCoreProducer.cc as done in #30209 

For tests, the following can provide a minimal setup
```
runTheMatrix.py -l 23434.21  -w upgrade -i all  --command=" -n 50 --customise Validation/Performance/TimeMemoryInfo.py --lazy_download" -t 4 >& pu.log &
```
A speedup visible with just Timing service (or FastTiming if preferred) can be enough to show speedup. Igprof analysis for update code is welcome as well.

@afiqaize @SohamBhattacharya  @guitargeek

**Comment by slava77** [comment]
assign reconstruction,upgrade

**Comment by guitargeek** [comment]
Hi @slava77, I actually wrote code to address this issue but then abandoned it due to lack of time. I made a PR with the code. Maybe you can test it with the bot to see if you can somehow use what I did?

**Comment by slava77** [comment]
> Hi @slava77, I actually wrote code to address this issue but then abandoned it due to lack of time. I made a PR with the code. Maybe you can test it with the bot to see if you can somehow use what I did?

@guitargeek 
thank you for the very quick follow up with #30956

**Comment by kpedro88** [comment]
+upgrade

**Comment by slava77** [comment]
+1

resolved via #30956 (merged Aug 29)
